### ISSUE TITLE
rocket_contrib 0.4.3 fails to build on latest nightly

### ISSUE THREAD

**Issue by RalfJung** [SergioBenitez/Rocket#1251]
Since the upgrade to latest nightly, building rocket_contrib fails:

```
error[E0463]: can't find crate for `notify`
  --> /home/r/.cargo/registry/src/github.com-1ecc6299db9ec823/rocket_contrib-0.4.3/src/templates/fairing.rs:35:5
   |
35 |     extern crate notify;
   |     ^^^^^^^^^^^^^^^^^^^^ can't find crate
```

The relevant part of my `Cargo.toml`:
```
rocket_contrib = { version = "0.4", features = ["handlebars_templates", "diesel_postgres_pool"] }
```

This is using the following versions of the rocket crate:
```
2605:"checksum rocket 0.4.3 (registry+https://github.com/rust-lang/crates.io-index)" = "0aa9ab5fad4a9b366ae884f5b5461e166fe8f94d23a561a91dfaab046c2f4fde"
2606:"checksum rocket_codegen 0.4.3 (registry+https://github.com/rust-lang/crates.io-index)" = "c56c540f73ae2291e8c1e4f9757f92fec7015c09fd90d7bb70349386a111602c"
2607:"checksum rocket_contrib 0.4.3 (registry+https://github.com/rust-lang/crates.io-index)" = "66926894ef94396204c53885f54a1b0e94a88ce6c9478c2fd403261fc0efb523"
2608:"checksum rocket_contrib_codegen 0.4.3 (registry+https://github.com/rust-lang/crates.io-index)" = "9bd54196728b177f2b5ef79a5e9766765a4bfc09f61d7f4c02a7604fe5f194a7"
2609:"checksum rocket_http 0.4.3 (registry+https://github.com/rust-lang/crates.io-index)" = "968a07cbfa9f8c8a0239aba359b80e1281e85da9415795c68f3ddeb5e1d636d6"
```
and rustc:
```
rustc 1.43.0-nightly (4ad624882 2020-03-03)
```

**Comment by jebrosen** [comment]
Upstream bug: rust-lang/cargo#7966.

See also https://github.com/SergioBenitez/Rocket/issues/1058#issuecomment-576051057.

Since `[target.cfg(debug_assertions).dependencies]` didn't actually do anything, perhaps we should backport commit 4151cd4 to 0.4.

**Comment by RalfJung** [comment]
> perhaps we should backport commit 4151cd4 to 0.4.

That seems like the easiest way forward. While the regression is caused by an upstream change, arguably the new upstream behavior is no worse than the current one.

**Comment by RalfJung** [comment]
Submitted PR: https://github.com/SergioBenitez/Rocket/pull/1255

**Comment by SergioBenitez** [comment]
Fixed by 2b9c4197d8abe805a45a08def07c845a3316148d.

**Comment by ThibaudDauce** [comment]
@SergioBenitez When a new version will be released? Or is there a workaround in user land to develop with Rocket with the current nightly? Maybe I can install a previous nightly?
